### PR TITLE
Add Chef 12 Support, Breaks Chef 11.

### PR DESCRIPTION
**Pull request by luckymike**

### CODE REVIEW

**Comment by luckymike** [comment]
Ready for Review!

**Comment by chrisroberts** [comment]
OMG

**Comment by chrisroberts** [comment]
Should remove the `Gemfile.lock` and add entry in `.gitignore`

File: attributes/default.rb [hunk 1]
```diff
@@ -34,3 +34,25 @@
...
+  :org_name => 'inception_llc',
+  :full_name => 'Chef Inception Organization',
+  :validator_pub_key => 'validator_pub.pem'
+}
+
+default[:chef_server_populator][:solo_org_user] = {
+  :name => 'populator',
+  :first => 'Populator',
+  :last => 'User',
+  :email => 'pop@hw-ops.com',
```
Comment (chrisroberts) [inline attributes/default.rb]: set `pop@hw-ops.com` to `pop@example.com` so if it ever actually does get used, it's using a reserved domain.

File: recipes/client.rb [hunk 2]
```diff
@@ -23,49 +21,109 @@
       types = [item['chef_server'].fetch('type', 'client')].flatten
       admin = item['chef_server'].fetch('admin', true)
       password = item['chef_server'].fetch('password', SecureRandom.urlsafe_base64(23))
+      full_name = item.fetch('full_name', client.capitalize)
+      first_name = full_name.split(' ').first
+      last_name = full_name.split(' ').last
+      email = item.fetch('email', "#{client}@null.com")
```
Comment (chrisroberts) [inline recipes/client.rb]: use `example.com` in place of `null.com`

File: recipes/restore.rb [hunk 3]
```diff
@@ -19,95 +19,86 @@
...
 
-file '/etc/chef-server/restore.json' do
+file '/etc/opscode/restore.json' do
   content JSONCompat.to_json_pretty(
-    :date => Time.now.to_i,
-    :file => file
-  )
+                                    :date => Time.now.to_i,
+                                    :file => file
+                                    )
```
Comment (chrisroberts) [inline recipes/restore.rb]: :see_no_evil:

Comment (luckymike) [inline recipes/restore.rb]: My emacs file is terrible!

**Comment by chrisroberts** [comment]
OMG OMG OMG

**Pull request merged**